### PR TITLE
TEG and APC Feedback

### PR DESCRIPTION
**Pull request by Azrun**
[QoL]
## About the PR 
* One Minute of Power Remaining: Red Lights on TEG, Red Point Light, and KLAXON BLARES!
* Insufficient molar quantity of gas at inlet (desired pressure would not provide 0.0005 moles of GAS): Red Lights on TEG, Red Point Light, and KLAXON BLARES!
* Five Minutes of Power Reaming: Yellow Lights on TEG, Yellow Point Light.

teg_warning_lights_a

Indication that circulator blowers are exceeding `perapc` and there is no surplus. (Sufficient power usage to cause drain on cell without anything else on the bus).
teg_warning_lights_b

Lights are per side BUT always indicates highest warning state.
 * Left exceeding charge rate while Right pulling enough for 1 minute left = ALL RED
 * Right exceeding charge  rate = Right side flashing yellow

## Why's this needed? 
Provides visual and auditory cues as to how much power TEG is using from blower.

```
(u)Azrun:
(*)TEG now has visual indicators for power draw on blower. Yellow indicates 5 minutes of power while two red lights and blaring klaxon means you have one minute remaining!  Single yellow can also indicate that blower power draw is enough to cause the APC power cell to drain.
```

### CODE REVIEW

File: code/WorkInProgress/TempEngine.dm [hunk 1]
```diff
@@ -733,6 +818,10 @@ datum/pump_ui/circulator_ui
 				running = 1
 			SPAWN_DBG(0.5 SECONDS)
 				spam_limiter = 0
+		else if(warnings > WARNING_5MIN && !(src.status & (BROKEN | NOPOWER)))
+			// Allow for klaxon to trigger when off cooldown if updateicon() not called
+			if(!ON_COOLDOWN(src, "klaxon", 5 SECOND))
```
Comment (ZeWaka) [inline code/WorkInProgress/TempEngine.dm:823]: 5 seconds could get annoying, but i guess it's supposed to be?

File: code/WorkInProgress/TempEngine.dm [hunk 2]
```diff
@@ -156,6 +172,7 @@
 		//Calculate necessary moles to transfer using PV = nRT
 		var/pressure_delta = (input_starting_pressure - output_starting_pressure)/2
 
+		src.warning_active = 0
```
Comment (ZeWaka) [inline code/WorkInProgress/TempEngine.dm:175]:
```suggestion
		src.warning_active = FALSE
```

**Review by ZeWaka — APPROVED** [review]

File: code/WorkInProgress/TempEngine.dm (hunk 2 above)
Comment (Azrun) [reply code/WorkInProgress/TempEngine.dm:175]: MIXED FEELINGS! 😱

File: code/WorkInProgress/TempEngine.dm (hunk 1 above)
Comment (Azrun) [reply code/WorkInProgress/TempEngine.dm]: Seemed fine while I was capturing screenshots and running around the engine room but decided to try to do other things while alarm was going and started to go insane.

Variable pitch was also just problematic as low end of varying pitch would stretch clip out to approximately 5 seconds. 😞 

It is enough to be somewhat jarring and a more consistent pitch feels much cleaner.

File: code/WorkInProgress/TempEngine.dm (hunk 2 above)
Comment (ZeWaka) [reply code/WorkInProgress/TempEngine.dm]: u put it as FALSE before!

Comment (Azrun) [reply code/WorkInProgress/TempEngine.dm]: Agreed.
I think that was a holdover from initial prototype with a binary indication. Setting a bit field to false felt odd, but the variable naming and some usage is more self documenting this way.

I lean more towards the reviewer in general to produce an artifact that is understandable by... not me.  👍

**Pull request merged**